posthog/batch_exports/http.py

```diff
@@ -142,25 +144,30 @@ def retry(self, *args, **kwargs) -> response.Response:
         batch_export_run = self.get_object()
 
         temporal = sync_connect()
-        backfill_id = backfill_export(
+        backfill_workflow_id = backfill_export(
             temporal,
             str(batch_export_run.batch_export.id),
             self.team_id,
             batch_export_run.data_interval_start,
             batch_export_run.data_interval_end,
         )
 
-        return response.Response({"backfill_id": backfill_id})
+        return response.Response({"backfill_id": backfill_workflow_id})
 
     @action(methods=["POST"], detail=True, required_scopes=["batch_export:write"])
     def cancel(self, *args, **kwargs) -> response.Response:
         """Cancel a batch export run."""
 
         batch_export_run: BatchExportRun = self.get_object()
 
-        temporal = sync_connect()
-        # TODO: check status of run beforehand
-        cancel_running_batch_export_run(temporal, batch_export_run)
+        if (
+            batch_export_run.status == BatchExportRun.Status.RUNNING
+            or batch_export_run.status == BatchExportRun.Status.STARTING
+        ):
+            temporal = sync_connect()
+            cancel_running_batch_export_run(temporal, batch_export_run)
```

Comment:
There is some race condition that could happen here (TOCTOU, basically) as the call to temporal may fail because the batch export is already cancelled even before our database has been updated. 

I don't think there is much we can (or should) do here, but we should `try/except` this to avoid a 500.

Reply:
good point 👍